tests/data/test_dataset.py

```diff
@@ -171,3 +171,18 @@ def test_lazy_load_until_access_full(tmp_path):
     assert dataset.attrs == {"molecule": 1, "hf_state": None}
     assert dataset.hf_state == 2
     assert dataset.attrs == {"molecule": 1, "hf_state": 2}
+
+
+def test_hamiltonian_is_loaded_properly(tmp_path):
+    """Test that __getattribute__ correctly loads hamiltonians from dicts."""
+    filename = str(tmp_path / "myset_hamiltonian.dat")
+    qml.data.Dataset._write_file(
+        {"terms": {"IIII": 0.1, "ZIII": 0.2}, "wire_map": {0: 0, 1: 1, 2: 2, 3: 3}}, filename
+    )
```

Comment:
I hadn't realized we could use `_write_file` to save single attributes like this. Nice

Reply:
I actually removed it before because it was only used by `write()`, but then I wanted it for tests so I put it back 😄 comes in clutch all the time now